feat(journal-crud): Basic Journal CRUD

d-e-v-esh opened the pull request:
## #54

This PR builds on the `journalResolver` and adds more functionalities. Now we can do basic CRUD operations with the journal entity. More will be added on top of it in forthcoming PRs.

### Changes

 - Updated Journal `model` and `typeDefs`
 - Added `getJournalByISSN` Query
 - Added `deleteJournal` Mutation
 - Added `updateJournal` Mutation

### Flags
 - Newly added journal fields `createdAt`, `updatedAt` and `createdBy` do not auto-populate yet. This will be added in future PRs.

CIPHERTron commented on src/models/Journal.js:
```diff
@@ -1,22 +1,54 @@
+    required: false,
+  },
+
+  policies: {
```
Hey, @d-e-v-esh we had also planned to have a `policyType` field right?

d-e-v-esh replied:
Shall I rename `policies` to `polityType`?

CIPHERTron commented on src/models/Journal.js:
```diff
@@ -1,22 +1,54 @@
+  },
+
+  policies: {
+    title: {
```
We can have this field as required i.e. `required: true,`

CIPHERTron commented on src/models/Journal.js:
```diff
@@ -1,22 +1,54 @@
+    },
+
+    firstYear: {
+      type: Number,
```
We can also have this field as required i.e. `required: true`

CIPHERTron commented on src/models/Journal.js:
```diff
@@ -1,22 +1,54 @@
+      type: Number,
+    },
+
+    lastYear: {
```
I think we can leave this field as not required if the journal doesn't have a last year

CIPHERTron reviewed (commented):
Overall PR looks good. Just a few nits and then it'll be good to go 🚀

CIPHERTron reviewed (commented):
> Shall I rename `policies` to `policyType`?

@d-e-v-esh Not really. `policyType` would be a property of `policies`

The `policies` object will have the following properties 👇🏻 

```javascript
policies {
title,
firstYear,
lastYear,
policyType
}
```

Out of which, only `lastYear` property can be named as `required:false` and rest as `required: true`

Also I noticed that you discarded the `lastYear` property from `policies` whereas I intended to say that we can leave as it is, without the `required: true` property. Please revert this change as well.

Lmk in-case of any queries.

d-e-v-esh committed:
feat(journal-crud): basic journal crud - #54
This commit contains the following:
 - Updated Journal model and typeDefs
 - getJournalByISSN Query
 - deleteJournal Mutation
 - updateJournal Mutation

d-e-v-esh commented:
@CIPHERTron 

There are actually more fields that are needed to be added under `policies` as said by Yo in the last meeting:
```js
policies {
  isDataAvailabilityStatementPublished: Boolean;
  isDataShared: Boolean;
  isDataPeerReviewed: Boolean;
  enforced: Boolean;
  enforcementEvidence: String;
}

```

After we merge user authentication into the main `dev` branch, I'll add these fields into the journal schema and go on towards integrating `users` and `journals`.

d-e-v-esh commented:
I updated it. Let me know if there are any other changes.

CIPHERTron reviewed (approved):
LGTM 🎉
